common-lib/src/main/scala/com/gu/mediaservice/lib/imaging/ImageOperations.scala

```diff
@@ -140,10 +140,39 @@ class ImageOperations(playPath: String) {
       transformSource = addImage(sourceFile)
       addOutput    = addDestImage(transformSource)(outputFile)
       _           <- runConvertCmd(addOutput, useImageMagick = sourceMimeType.contains(Tiff))
+      _           <- checkForOutputFileChange(outputFile)
+    } yield outputFile
+  }
+
+  // When a layered tiff is unpacked, the temp file (blah.something) is moved 
+  // to blah-0.something and contains the composite layer (which is what we want).
+  // Other layers are then saved as blah-1.something etc.
+  // As the file has been renamed, the file object still exists, but has the wrong name
+  // We will need to put it back where it is expected to be found, and clean up the other
+  // files. 
+  private def checkForOutputFileChange(f: File): Future[Unit] = Future {
+    val fileBits = f.getAbsolutePath.split("\\.").toList
+    val mainPart = fileBits.dropRight(1).mkString(".")
+    val extension = fileBits.last
+
+    // f2 is the blah-0 name that gets created from a layered tiff.
+    val f2 = new File(List(s"${mainPart}-0", extension).mkString("."))
+    if (f2.exists()) {
```

Comment:
Is it possible for `f2` to not exist here?

Reply:
Yes, absolutely,

Reply:
If the file is not layered, then it only makes the basic "blah.something" file.  It's only if it's layered that it creates blay-0, blah-1 etc.